### PR TITLE
Fix failing tests that use TranslateToOpenQASM pipeline

### PR DESCRIPTION
**Pull request by annagrin**
### Description
Fix failing tests that use `TranslateToOpenQASM` pipeline

- Update `cc.GlobalOp` with symbol visibility so it can be removed by `symbol-dce` pass if not used
  - set private visibility on various auxiliary global ops created during other passes.
- Run additional passes as part of pipelines that translate to OpenQasm2:
  - all passes related to unitary synthesis
  - decomposition patterns "CCzToCx"
  - add new decomposition patterns  "RxAdjToRx", "RyAdjToRy","RyAdjToRy",
  - symbol-dce
  - and more
 - Update bracket pipeline with missing passes as well
 - Add tests

Towards: https://github.com/NVIDIA/cuda-quantum/issues/2325
Note: a few tests from that issue are still failing:
- custom_operation_ctrl.cpp
- qspan_slices.cpp
- phase_estimation.cpp

### CODE REVIEW

File: include/cudaq/Optimizer/Transforms/Passes.td [hunk 1]
```diff
@@ -407,7 +407,7 @@ def GetConcreteMatrix : Pass<"get-concrete-matrix", "mlir::func::FuncOp"> {
           return
         }
 
-        cc.global constant @__nvqpp__mlirgen__function_foo_generator_1.bar.rodata_0 ((dense<[(0.000000e+00,0.000000e+00), (1.000000e+00,0.000000e+00), (1.000000e+00,0.000000e+00), (0.000000e+00,0.000000e+00)]> : tensor<4xcomplex<f64>>) : !cc.array<complex<f64> x 4>
+        cc.global constant @__nvqpp__mlirgen__function_foo_generator_1.bar.rodata_0 ((dense<[(0.000000e+00,0.000000e+00), (1.000000e+00,0.000000e+00), (1.000000e+00,0.000000e+00), (0.000000e+00,0.000000e+00)]> : tensor<4xcomplex<f64>>) : !cc.array<complex<f64> x 4> {sym_visibility = "private"}
```
Comment (schweitzpgi) [inline include/cudaq/Optimizer/Transforms/Passes.td:410]: Can we make this pretty-print nicer?
```
cc.global constant private @__nvqpp__mlirgen_function....
```

File: lib/Optimizer/Transforms/DecompositionPatterns.cpp [hunk 2]
```diff
@@ -1181,6 +1181,41 @@ struct RxToPhasedRx : public OpRewritePattern<quake::RxOp> {
   }
 };
 
+// quake.rx<adj> (θ) target
+// ─────────────────────────────────
+// quake.rx(-θ) target
```
Comment (schweitzpgi) [inline lib/Optimizer/Transforms/DecompositionPatterns.cpp:1186]: These adjoint forms are a bit surprising. Do you know what is generating them? The code that generates generalized adjoints doesn't use these. It just negates the angle argument.

I think we should probably get rid of these `r?<adj>` as a valid operation. We did that with `x<adj>` and such.

File: targettests/execution/custom_operation_adj.cpp [hunk 3]
```diff
@@ -6,12 +6,15 @@
...
  ******************************************************************************/
 
-// RUN: nvq++ %cpp_std --enable-mlir %s -o %t && %t | FileCheck %s
-// RUN: nvq++ %cpp_std --target anyon --emulate %s -o %t && %t | FileCheck %s
-// RUN: nvq++ %cpp_std --target ionq --emulate %s -o %t && %t | FileCheck %s
-// RUN: nvq++ %cpp_std --target iqm --emulate --iqm-machine Apollo %s -o %t && %t | FileCheck %s
-// RUN: nvq++ %cpp_std --target oqc --emulate %s -o %t && %t | FileCheck %s
-// RUN: nvq++ %cpp_std --target quantinuum --emulate %s -o %t && %t | FileCheck %s
+// clang-format off
+// RUN: nvq++ -std=c++17 --enable-mlir                             %s -o %t && %t | FileCheck %s
```
Comment (schweitzpgi) [inline targettests/execution/custom_operation_adj.cpp:10]: I find all this whitespace sort of annoying for cut-n-paste reasons.

File: lib/Optimizer/Transforms/DecompositionPatterns.cpp (hunk 2 above)
Comment (annagrin) [reply lib/Optimizer/Transforms/DecompositionPatterns.cpp:1186]: I also found that some decomposition patterns create them, "CRxToCX" for example

Comment (annagrin) [reply lib/Optimizer/Transforms/DecompositionPatterns.cpp:1186]: chatted offline, will postpone the `adjoint` removal for later

**Review by schweitzpgi — APPROVED** [review]
Thanks, Anna.

**Pull request merged**